### PR TITLE
Implement a related facts component

### LINKED ISSUE #79: Related facts
We were looking at having a list of related facts (probably 3-5 of them) under each large fact card. This plugin may be the way to go: https://www.gatsbyjs.org/packages/gatsby-remark-related-posts/

### PR DESCRIPTION
**Pull request by UtkarshKr007**
We are using [gatsby-remark-related-posts](https://www.gatsbyjs.org/packages/gatsby-remark-related-posts/) plugin which calculates the similarity between fact texts using [tf-idf](https://en.wikipedia.org/wiki/Tf%E2%80%93idf) and [Cosine similarity](https://en.wikipedia.org/wiki/Cosine_similarity). Our job is to query, limit, sort and render the posts inside a component that can be referenced in fact template and displayed under every fact page.

### CODE REVIEW

**Comment by floogulinc** [comment]
I don't know if this way of finding related facts is what we want considering we are adding tags.

**Comment by UtkarshKr007** [comment]
I believe we thought it would be interesting to see what facts seem to be related by their text similarity albeit having different categories/tags. I personally didn't notice any discernible pattern of certain facts being related to others while totally different from others, but that's probably in part to the low count of facts we have. 
If we scrap this method and implement one that groups related facts by the number of tags they share or something similar, perhaps we could leave this branch alone and pull master into it later to see how this plugin works given a larger amount of facts.

File: src/components/horizontal-fact-card.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        <Card className={classes.relatedCard}>
+            <CardActionArea to={factSlug} className={classes.actionArea}>
+                <CardMedia
+                    component={Img}
+                    fixed={factImage.src.childImageSharp.fixed}
+                    alt={factImage.alt}
+                />
+
+                <CardContent>
+                    <Typography component="h6" variant="h6">
```
Comment (floogulinc) [inline src/components/horizontal-fact-card.js]:
```suggestion
                    <Typography component="h3" variant="h6">
```

File: src/components/horizontal-fact-card.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                        {factTitle}
+                    </Typography>
+                    <Typography>
```
Comment (floogulinc) [inline src/components/horizontal-fact-card.js:35]: This should be variant `body2` I think.

File: src/components/horizontal-fact-card.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { makeStyles } from '@material-ui/core/styles';
+import Card from '@material-ui/core/Card';
+import { CardActionArea } from 'gatsby-theme-material-ui'
+import CardMedia from '@material-ui/core/CardMedia';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+
+const useStyles = makeStyles({
+    relatedCard: {
+        marginBottom: '5px',
```
Comment (floogulinc) [inline src/components/horizontal-fact-card.js:12]: This margin should be handled in the component using it and should be more like `16px`

File: src/templates/factTemplate.js [hunk 4]
```diff
@@ -22,24 +24,28 @@ export default ({ data }) => {
...
+      <div className={classes.pageWidth}>
+        <LargeFactCard 
+          className={classes.card} 
+          factTitle={fact.frontmatter.title} 
+          factImage={fact.frontmatter.image} 
+          factSource={fact.frontmatter.source} 
+          factCategory={fact.frontmatter.category} 
+          factHTML={fact.html} 
+        />
+        <h2>Related Facts</h2>
```
Comment (floogulinc) [inline src/templates/factTemplate.js]: This should be a `Typography` component.

File: src/templates/factTemplate.js [hunk 5]
```diff
@@ -73,11 +79,22 @@ export const query = graphql`
...
+          excerpt(pruneLength: 50)
           fields {
             slug
           }
           frontmatter {
             title
+            image {
+              src {
+                childImageSharp {
+                  fixed(width: 100, height: 100) {
```
Comment (floogulinc) [inline src/templates/factTemplate.js:91]: We decided to try having this image be `fluid` with a max width of `100`.

**Review by floogulinc — CHANGES_REQUESTED** [review]

**Comment by floogulinc** [comment]
Are the related facts being ordered by relatedness? They all seem to be the same 3-4.

File: src/templates/factTemplate.js [hunk 6]
```diff
@@ -1,40 +1,56 @@
...
+      <div className={classes.pageWidth}>
+        <LargeFactCard
+          className={classes.card}
+          factTitle={fact.frontmatter.title}
+          factImage={fact.frontmatter.image}
+          factSource={fact.frontmatter.source}
+          factCategory={fact.frontmatter.category}
+          factHTML={fact.html}
+        />
+        <Typography variant="h6" component="h3" className={classes.relatedFactsTitle} > Related Facts </Typography>
```
Comment (floogulinc) [inline src/templates/factTemplate.js]:
```suggestion
        <Typography variant="h6" component="h3" className={classes.relatedFactsTitle}>Related Facts</Typography>
```

**Review by floogulinc — APPROVED** [review]
Looks like you need to fix the merge conflicts with master

**Pull request merged**